packages/openid4vc/src/openid4vc-holder/OpenId4vcSiopHolderService.ts

```diff
@@ -111,27 +120,65 @@ export class OpenId4VcSiopHolderService {
         throw new CredoError("Unable to extract 'response_uri' from authorization request")
       }
 
-      const { verifiablePresentations, presentationSubmission } =
-        await this.presentationExchangeService.createPresentation(agentContext, {
-          credentialsForInputDescriptor: presentationExchange.credentials,
-          presentationDefinition: authorizationRequest.presentationDefinitions[0].definition,
+      if (authorizationRequest.presentationDefinitions && authorizationRequest.presentationDefinitions.length > 0) {
+        if (!presentationExchange) {
+          throw new CredoError(
+            'Authorization request included presentation definition. `presentationExchange` MUST be supplied to accept authorization requests.'
+          )
+        }
+
+        const { verifiablePresentations, presentationSubmission } =
+          await this.presentationExchangeService.createPresentation(agentContext, {
+            credentialsForInputDescriptor: presentationExchange.credentials,
+            presentationDefinition: authorizationRequest.presentationDefinitions[0].definition,
+            challenge: nonce,
+            domain: clientId,
+            presentationSubmissionLocation: DifPresentationExchangeSubmissionLocation.EXTERNAL,
+            openid4vp: {
+              mdocGeneratedNonce: authorizationResponseNonce,
+              responseUri,
+            },
+          })
+
+        if (wantsIdToken && !openIdTokenIssuer) {
+          openIdTokenIssuer = this.getOpenIdTokenIssuerFromVerifiablePresentation(verifiablePresentations[0])
+        }
+
+        presentationExchangeOptions = {
+          verifiablePresentations: verifiablePresentations.map((vp) => getSphereonVerifiablePresentation(vp)),
+          presentationSubmission,
+          vpTokenLocation: VPTokenLocation.AUTHORIZATION_RESPONSE,
+        }
+      } else {
+        if (!dcql) {
+          throw new CredoError(
+            'Authorization request included dcql query. `dcql` MUST be supplied to accept authorization requests.'
+          )
+        }
+
+        const presentationRecord = await this.dcqlService.createPresentationRecord(agentContext, {
+          credentialQueryToCredential: dcql.credentials,
           challenge: nonce,
           domain: clientId,
-          presentationSubmissionLocation: DifPresentationExchangeSubmissionLocation.EXTERNAL,
           openid4vp: {
             mdocGeneratedNonce: authorizationResponseNonce,
             responseUri,
           },
         })
 
-      presentationExchangeOptions = {
-        verifiablePresentations: verifiablePresentations.map((vp) => getSphereonVerifiablePresentation(vp)),
-        presentationSubmission,
-        vpTokenLocation: VPTokenLocation.AUTHORIZATION_RESPONSE,
-      }
+        dcqlOptions = {
+          encodedPresentationRecord: await this.dcqlService.getEncodedPresentationRecord(presentationRecord),
```

Comment:
Why is it called presentation record?

Reply:
I haven't often seen the type in the name if you can infer it from the type. I think `encodedPresentations` would be just as clear, while more brief?